src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -665,6 +755,27 @@ private List<SyntaxNode> VerifyAndPermuteParamNodes(IEnumerable<XmlElementSyntax
         }
 
         protected override IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document)
-            => SpecializedCollections.SingletonEnumerable(new ChangeSignatureFormattingRule()).Concat(Formatter.GetDefaultFormattingRules(document));
+            => Formatter.GetDefaultFormattingRules(document).Concat(new ChangeSignatureFormattingRule());
+
+        protected override SyntaxNode AddNameToArgument(SyntaxNode newArgument, string name)
+        {
+            return (newArgument as ArgumentSyntax).WithNameColon(NameColon(name));
+        }
+
+        protected override SyntaxNode CreateExplicitParamsArrayFromIndividualArguments(SeparatedSyntaxList<SyntaxNode> newArguments, int indexInExistingList, IParameterSymbol parameterSymbol)
+        {
+            var listOfArguments = SeparatedList(newArguments.Skip(indexInExistingList).Select(a => ((ArgumentSyntax)a).Expression), newArguments.GetSeparators().Skip(indexInExistingList));
```

Comment:
1. by stripping off the Expression of an argument, aren't we droping things like `ref/out`?
2. it would be nice to be strongly typed here.

Reply:
These are individual elements of a params array, which I don't think can have modifiers?

Reply:
gotcha.  can you add a comment here that this is ok?

Reply:
Yep! Done.